protocols/gossipsub/src/protocol.rs

```diff
@@ -59,20 +59,20 @@ impl ProtocolConfig {
     ///
     /// Sets the maximum gossip transmission size.
     pub fn new(
-        id_prefix: Cow<'static, str>,
+        id: Cow<'static, str>,
         max_transmit_size: usize,
         validation_mode: ValidationMode,
         support_floodsub: bool,
+        id_is_prefix: bool,
     ) -> ProtocolConfig {
-        // support version 1.1.0 and 1.0.0 with user-customized prefix
         let mut protocol_ids = vec![
-            ProtocolId::new(id_prefix.clone(), PeerKind::Gossipsubv1_1),
-            ProtocolId::new(id_prefix, PeerKind::Gossipsub),
+            ProtocolId::new(id.clone(), PeerKind::Gossipsubv1_1, id_is_prefix),
+            ProtocolId::new(id, PeerKind::Gossipsub, id_is_prefix),
```

Comment:
This is kind of what I was saying. We need to be explicit about what version a custom id will have. 

The way you have currently implemented this, the node will advertise two protocol ids (assuming `id_is_prefix` is false). 
- `id` - which matches `Gossipsubv1_1` and
- `id` - which matches `Gossipsub`. 

When we see another node on the network with this custom id, it will just match the first one and therefore we will assume all nodes are 1.1 nodes. If there is a 1.0 node on the network, we may send it messages it doesn't understand and they will treat these as invalids. 

To be explicit, perhaps the config should allow a user to provide a `Vec<ProtocolId>` which can map which node versions match to which id. 
If you just want to assume every single node on the network is 1.1 (which is likely to cause issues when 1.2 comes) then maybe the config parameter can be something like `protocol_id_v1_1` and we disable support for all other versions (this is done by just advertising this protocol_id and leaving everything else unsupported).